models/superrare/superrare_ethereum_events.sql

```diff
@@ -0,0 +1,372 @@
+        case when a.`_currencyAddress` = '0xba5bde662c17e2adff1075610382b9b691296350'
+                then ROUND((3*((a.`_amount`/1e18)*average_price_that_day_eth_per_rare*ep.price)/100),7)
+            else ROUND((3*((a.`_amount`/1e18)*ep.price)/100),7) end as platform_fee_amount_usd,
+        '3' as platform_fee_percentage,
+
+        -- SuperRare royalty fee notes:
+        -- This query aggregates both primary sales and secondary sales on the platform
+        -- Primary sales -- 85% of listed sale price goes to the artist
+        -- Secondary sales -- 10% royalty on listed sale price goes to the artist
+        -- For simplicity and to align with other marketplaces, this code shows a 10% royalty for all secondary sales 
+
+        case  when ma721.mint_sent_to = po.previous_owner then 'Primary' -- auctions
+                when ma721.mint_sent_to = `_seller` then 'Primary'
+                when ma20.mint_address = `_seller` then 'Primary'
+            else 'Secondary' end as superrare_sale_type,
+
+        case when ma721.mint_sent_to != po.previous_owner and ma721.mint_sent_to != `_seller` and ma20.mint_address != `_seller` -- secondary sale 
+                then ROUND((10*(a.`_amount`)/100),7) 
+            else null end as royalty_fee_amount_raw,
+        case when ma721.mint_sent_to != po.previous_owner and ma721.mint_sent_to != `_seller` and ma20.mint_address != `_seller` -- secondary sale 
+                then ROUND((10*((a.`_amount`/1e18))/100),7)
+            else null end as royalty_fee_amount,
+        case when a.`_currencyAddress` = '0xba5bde662c17e2adff1075610382b9b691296350' and ma721.mint_sent_to != po.previous_owner and ma721.mint_sent_to != `_seller` and ma20.mint_address != `_seller` -- secondary sale and rare
+                then ROUND((10*((a.`_amount`/1e18)*average_price_that_day_eth_per_rare*ep.price)/100),7)
+             when ma721.mint_sent_to != po.previous_owner and ma721.mint_sent_to != `_seller` and ma20.mint_address != `_seller` -- secondary sales 
+                then ROUND((10*((a.`_amount`/1e18)*ep.price)/100),7)
+            else null end as royalty_fee_amount_usd,
+        '10' as royalty_fee_percentage,
+        case when ma721.mint_sent_to is not null 
+                then ma721.mint_sent_to
+            else ma20.mint_address end as royalty_fee_receive_address,
+        case when a.`_currencyAddress` = '0xba5bde662c17e2adff1075610382b9b691296350'
+                then 'RARE'
+            else 'ETH' -- only RARE and ETH possible 
+            end as royalty_fee_currency_symbol,
+        'superrare' || '-' || a.evt_tx_hash || '-' ||  a.`_tokenId`::string || '-' ||  `_seller`::string || '-' || COALESCE(a.contract_address) || '-' || 'Trade'  as unique_trade_id
+
+from all_superrare_sales a
+
+left outer join (select minute, price 
+                -- from prices.usd 
+                from {{ source('prices','usd') }}
+                where blockchain = 'ethereum' and symbol = 'WETH' ) ep 
```

Comment:
where blockchain = 'ethereum' and symbol = 'WETH'
                {% if is_incremental() %} -- this filter will only be applied on an incremental run
                and minute >= date_trunc("day", now() - interval '1 week')
                {% endif %}) ep